fdbclient/common.go

```diff
@@ -83,24 +90,29 @@ func parseMachineReadableStatus(logger logr.Logger, contents []byte, checkForPro
 	return status, nil
 }
 
-// getFDBDatabase opens an FDB database.
+// getFDBDatabase returns the singleton FDB database. May return an error if initializing the singleton failed.
 func getFDBDatabase(cluster *fdbv1beta2.FoundationDBCluster) (fdb.Database, error) {
-	clusterFile, err := createClusterFile(cluster)
-	if err != nil {
-		return fdb.Database{}, err
-	}
+	databaseSingletonMutex.Lock()
+	defer databaseSingletonMutex.Unlock()
+	if databaseSingleton[cluster.UID] == nil {
+		clusterFile, err := createClusterFile(cluster)
+		if err != nil {
+			return fdb.Database{}, err
+		}
 
-	database, err := fdb.OpenDatabase(clusterFile)
-	if err != nil {
-		return fdb.Database{}, err
-	}
+		database, err := fdb.OpenDatabase(clusterFile)
```

Comment:
You could use `OpenWithConnectionString` to bypass the other cache on Go SDK side; was briefly discussed with @johscheuer [here](https://github.com/FoundationDB/fdb-kubernetes-operator/issues/2171#issuecomment-2552009785)

Reply:
`OpenWithConnectionString` is not available in the 7.1 bindings, so the operator cannot make use of this method.